## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-xl.sh | :x: [more...](#d18ea325-1396-4b13-9a9f-ed5ecf6a68ce) |
| mint-large-bucket.sh | :x: [more...](#0d7c3dbe-52d0-4895-883d-27a110feaed6) |
| mint-dist-xl.sh | :x: [more...](#8bd08ca2-8960-4b39-b12d-c943fe782907) |
| mint-gateway-s3.sh | :x: [more...](#7d156e3d-21b6-4c94-aebc-023f2f42e6a5) |
| mint-gateway-azure.sh | :x: [more...](#20dd3ac0-4de8-4613-a4a6-8abfd10765f6) |
| mint-zoned.sh | :x: [more...](#2cdd1575-73c8-4050-a94f-64d3a2cc5f0a) |
<a name="2cdd1575-73c8-4050-a94f-64d3a2cc5f0a"></a>
### 9981-bc0a58e/mint-zoned.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:31566
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 1d7d38c5f092:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 18 seconds
(2/15) Running aws-sdk-java tests ... done in 4 seconds
(3/15) Running aws-sdk-php tests ... done in 48 seconds
(4/15) Running aws-sdk-ruby tests ... done in 10 seconds
(5/15) Running awscli tests ... done in 2 minutes and 51 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 1 minutes and 44 seconds
(8/15) Running minio-dotnet tests ... done in 3 minutes and 14 seconds
(9/15) Running minio-go tests ... FAILED in 3 minutes and 35 seconds
{
  "args": {
    "bucketName": "minio-go-test-jd3w156ke6q1wp6s",
    "objectName": "wlywywrnpbqr52oxrxdpu9hzu44afd",
    "objectPrefix": "",
    "recursive": "true"
  },
  "duration": 5155,
  "error": "The bucket you tried to delete is not empty. You must delete all versions in the bucket.",
  "function": "RemoveObjectsWithOptions(bucketName, objectsCh, opts)",
  "message": "Cleanup failed",
  "name": "minio-go: testRemoveObjectsWithOptions",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 7 minutes and 2 seconds
(10/15) Running minio-js tests ... done in 58 seconds
(11/15) Running minio-py tests ... done in 4 minutes and 20 seconds
(12/15) Running s3cmd tests ... done in 25 seconds
(13/15) Running s3select tests ... done in 11 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="7d156e3d-21b6-4c94-aebc-023f2f42e6a5"></a>
### 9981-bc0a58e/mint-gateway-s3.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:30112
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp f50ddcbf91e4:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 0 seconds
(2/15) Running aws-sdk-java tests ... done in 3 seconds
(3/15) Running aws-sdk-php tests ... done in 41 seconds
(4/15) Running aws-sdk-ruby tests ... done in 2 seconds
(5/15) Running awscli tests ... done in 2 minutes and 0 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 25 seconds
(8/15) Running minio-dotnet tests ... done in 31 seconds
(9/15) Running minio-go tests ... done in 29 seconds
(10/15) Running minio-java tests ... done in 39 seconds
(11/15) Running minio-js tests ... FAILED in 30 minutes and 11 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 16,
  "status": "FAIL",
  "error": "S3Error: The bucket you tried to delete is not empty at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(11/15) Running minio-py tests ... done in 54 seconds
(12/15) Running s3cmd tests ... done in 59 seconds
(13/15) Running s3select tests ... done in 3 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="d18ea325-1396-4b13-9a9f-ed5ecf6a68ce"></a>
### 9981-bc0a58e/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev6.minio.io:31493
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp c16c8b2c56fd:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 3 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 45 seconds
(4/15) Running aws-sdk-ruby tests ... done in 4 seconds
(5/15) Running awscli tests ... done in 2 minutes and 14 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 30 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 8 seconds
(9/15) Running minio-go tests ... FAILED in 1 minutes and 26 seconds
{
  "args": {
    "bucketName": "minio-go-test-e2sszob2h691x55y",
    "objectName": "rf5z5omeujfp0j0m2r254nh3epzomx",
    "objectPrefix": "",
    "recursive": "true"
  },
  "duration": 3132,
  "error": "The bucket you tried to delete is not empty. You must delete all versions in the bucket.",
  "function": "RemoveObjectsWithOptions(bucketName, objectsCh, opts)",
  "message": "Cleanup failed",
  "name": "minio-go: testRemoveObjectsWithOptions",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 3 minutes and 14 seconds
(10/15) Running minio-js tests ... done in 53 seconds
(11/15) Running minio-py tests ... done in 6 minutes and 38 seconds
(12/15) Running s3cmd tests ... done in 22 seconds
(13/15) Running s3select tests ... done in 10 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
<a name="0d7c3dbe-52d0-4895-883d-27a110feaed6"></a>
### 9981-bc0a58e/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev7.minio.io:31418
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 8ce682469481:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 14 seconds
(2/15) Running aws-sdk-java tests ... FAILED in 3 seconds
Jul 07, 2020 7:42:38 AM com.amazonaws.util.Base64 warn
WARNING: JAXB is unavailable. Will fallback to SDK implementation which may be less performant.If you are using Java 9+, you will need to include javax.xml.bind:jaxb-api as a dependency.
Exception in thread "main" com.amazonaws.services.s3.model.AmazonS3Exception: Server not initialized, please try again. (Service: Amazon S3; Status Code: 503; Error Code: XMinioServerNotInitialized; Request ID: 161F688FA97C9A56; S3 Extended Request ID: 2c2417f8-2d7f-41d6-af92-295280959e29), S3 Extended Request ID: 2c2417f8-2d7f-41d6-af92-295280959e29
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1742)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleServiceErrorResponse(AmazonHttpClient.java:1371)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1347)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1127)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:784)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:752)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:726)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:686)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:668)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:532)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:512)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:5052)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4998)
	at com.amazonaws.services.s3.AmazonS3Client.invoke(AmazonS3Client.java:4992)
	at com.amazonaws.services.s3.AmazonS3Client.listObjects(AmazonS3Client.java:895)
	at com.amazonaws.services.s3.AmazonS3Client.listObjects(AmazonS3Client.java:863)
	at io.minio.awssdk.tests.FunctionalTests.teardown(Unknown Source)
	at io.minio.awssdk.tests.FunctionalTests.main(Unknown Source)
(2/15) Running aws-sdk-php tests ... FAILED in 6 seconds
{
  "name": "aws-sdk-php",
  "function": "anonDeleteObjects ( array $params = [] )",
  "args": {
    "Bucket": "aws-sdk-php-5f04276f79ae1",
    "Object": "obj1"
  },
  "duration": "420",
  "status": "FAIL",
  "error": "Error executing \"DeleteObjects\" on \"http://minio-dev7.minio.io:31418/aws-sdk-php-5f04276f79ae1?delete\"; AWS HTTP error: Server error: `POST http://minio-dev7.minio.io:31418/aws-sdk-php-5f04276f79ae1?delete` resulted in a `503 Service Unavailable` response:\n<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>XMinioServerNotInitialized</Code><Message>Server not initialized, pl (truncated...)\n XMinioServerNotInitialized (server): Server not initialized, please try again. - <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>XMinioServerNotInitialized</Code><Message>Server not initialized, please try again.</Message><BucketName>aws-sdk-php-5f04276f79ae1</BucketName><Resource>/aws-sdk-php-5f04276f79ae1</Resource><RequestId>161F68912A4F9CB3</RequestId><HostId>2c2417f8-2d7f-41d6-af92-295280959e29</HostId></Error>"
}
(2/15) Running aws-sdk-ruby tests ... FAILED in 3 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 2126.79,
  "error": "Server not initialized, please try again.",
  "status": "FAIL"
}
(2/15) Running awscli tests ... FAILED in 14 seconds
{
  "name": "awscli",
  "duration": 11796,
  "function": "delete_bucket\n",
  "status": "FAIL",
  "error": "fatal error: An error occurred (XMinioServerNotInitialized) when calling the ListObjectsV2 operation (reached max retries: 4): Server not initialized, please try again.remove_bucket failed: Unable to delete all objects in the bucket, bucket will not be deleted."
}
(2/15) Running healthcheck tests ... done in 0 seconds
(3/15) Running mc tests ... FAILED in 2 seconds
{
  "name": "mc",
  "duration": "281",
  "function": "test_put_object_0byte",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-30522 --quiet --no-color cp myminio/mc-test-bucket-32385/mc-test-object-2777 mc-test-object-2777.downloaded  >>> \nmc: <ERROR> Unable to validate source myminio/mc-test-bucket-32385/mc-test-object-2777"
}
(3/15) Running minio-dotnet tests ... FAILED in 16 seconds
{
  "name": "minio-dotnet : RemoveObject_Test1",
  "function": "Task RemoveObjectAsync(string bucketName, string objectName, CancellationToken cancellationToken = default(CancellationToken))",
  "args": {
    "bucketName": "miniodotnetcf506tm2m3yibc5",
    "objectName": "]&+@]}d%&&"
  },
  "duration": 8,
  "status": "FAIL",
  "alert": "",
  "message": "MinIO API responded with message=Server not initialized, please try again.",
  "error": "Minio.Exceptions.UnexpectedMinioException: MinIO API responded with message=Server not initialized, please try again.\n   at Minio.MinioClient.ParseErrorFromContent(IRestResponse response) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 538\n   at Minio.MinioClient.ParseError(IRestResponse response) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 431\n   at Minio.MinioClient.HandleIfErrorResponse(IRestResponse response, IEnumerable`1 handlers, DateTime startTime) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 605\n   at Minio.MinioClient.ExecuteTaskCoreAsync(IEnumerable`1 errorHandlers, IRestRequest request, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/MinioClient.cs:line 405\n   at Minio.MinioClient.MakeBucketAsync(String bucketName, String location, CancellationToken cancellationToken) in /q/.q/sources/minio-dotnet/Minio/ApiEndpoints/BucketOperations.cs:line 98\n   at Minio.Functional.Tests.FunctionalTest.Setup_Test(MinioClient minio, String bucketName) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 374\n   at Minio.Functional.Tests.FunctionalTest.RemoveObject_Test1(MinioClient minio) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 2011"
}
(3/15) Running minio-go tests ... FAILED in 1 minutes and 53 seconds
{
  "args": {
    "bucketName": "minio-go-test-fot5adpyw1nzgq0b",
    "fileName": "",
    "objectName": "",
    "opts": ""
  },
  "duration": 112520,
  "error": "Server not initialized, please try again.",
  "function": "FPutObject(bucketName, objectName, fileName, opts)",
  "message": "MakeBucket failed",
  "name": "minio-go: testFPutObjectV2",
  "status": "FAIL"
}
(3/15) Running minio-java tests ... FAILED in 7 minutes and 18 seconds
{
  "name": "minio-java",
  "function": "listenBucketNotification(ListenBucketNotificationArgs args)",
  "args": "prefix=prefix, suffix=suffix, events={\"s3:ObjectCreated:*\", \"s3:ObjectAccessed:*\"}",
  "duration": 995,
  "status": "FAIL",
  "error": "java.lang.IllegalArgumentException: unknown error code string 'XMinioServerNotInitialized' >>> [io.minio.Xml.unmarshal(Xml.java:68), io.minio.MinioClient.execute(MinioClient.java:1050), io.minio.MinioClient.putObject(MinioClient.java:7582), io.minio.MinioClient.putObject(MinioClient.java:4682), io.minio.MinioClient.putObject(MinioClient.java:4881), FunctionalTest.listenBucketNotification_test1(FunctionalTest.java:3320), FunctionalTest.runTests(FunctionalTest.java:3867), FunctionalTest.main(FunctionalTest.java:4056)]"
}
(3/15) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 3,
  "status": "FAIL",
  "error": "S3Error: Server not initialized, please try again. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:86:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:156:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:80:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (_stream_readable.js:673:10) at endReadableNT (_stream_readable.js:1201:12) at processTicksAndRejections (internal/process/task_queues.js:84:21)"
}
(3/15) Running minio-py tests ... FAILED in 17 minutes and 50 seconds
{
  "name": "minio-py:test_list_objects_with_1001_files",
  "function": "list_objects(bucket_name, prefix, recursive)",
  "args": {
    "bucket_name": "minio-py-test-dd41ed03-d63c-426f-a7d9-599c9972227c",
    "object_name": "eb4c276a-dfe8-4242-a183-8a9a205aa474_0 ~ eb4c276a-dfe8-4242-a183-8a9a205aa474_1000"
  },
  "duration": 1026940,
  "message": "HTTPConnectionPool(host='minio-dev7.minio.io', port=31418): Max retries exceeded with url: /minio-py-test-dd41ed03-d63c-426f-a7d9-599c9972227c/?delimiter=%2F&prefix= (Caused by ResponseError('too many 503 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1186, in test_list_objects_with_1001_files\n    list_objects_api_test(client, bucket_name, no_of_created_files)\n  File \"/mint/run/core/minio-py/tests.py\", line 1070, in list_objects_api_test\n    bucket_name, *argv))  # expect all objects to be listed\n  File \"/mint/run/core/minio-py/tests.py\", line 1056, in count_objects\n    for obj in objects:\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1036, in list_objects\n    headers=headers)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 2000, in _url_open\n    preload_content=preload_content)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 336, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 828, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 828, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 828, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 805, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 439, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-dev7.minio.io', port=31418): Max retries exceeded with url: /minio-py-test-dd41ed03-d63c-426f-a7d9-599c9972227c/?delimiter=%2F&prefix= (Caused by ResponseError('too many 503 error responses',))\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2093, in main\n    test_list_objects_with_1001_files(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1189, in test_list_objects_with_1001_files\n    raise Exception(err)\nException: HTTPConnectionPool(host='minio-dev7.minio.io', port=31418): Max retries exceeded with url: /minio-py-test-dd41ed03-d63c-426f-a7d9-599c9972227c/?delimiter=%2F&prefix= (Caused by ResponseError('too many 503 error responses',))\n",
  "status": "FAIL"
}
(3/15) Running s3cmd tests ... FAILED in 52 seconds
{
  "name": "s3cmd",
  "duration": "45382",
  "function": "setup",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 3 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 6 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 9 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 12 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 15 sec...\nERROR: S3 error: 503 (XMinioServerNotInitialized): Server not initialized, please try again."
}
(3/15) Running s3select tests ... FAILED in 10 minutes and 5 seconds
{
  "name": "s3select:test_sql_operators_precedence",
  "function": "select_object_content(bucket_name, object_name, opts)",
  "args": {
    "total_tests": 14,
    "total_success": 10
  },
  "duration": 600951,
  "message": "HTTPConnectionPool(host='minio-dev7.minio.io', port=31418): Max retries exceeded with url: /s3select-test-4f49adf8-e091-46ba-bbf2-8f07be81b716?location= (Caused by ResponseError('too many 503 error responses',))",
  "error": "Traceback (most recent call last):\n  File \"./tests.py\", line 54, in main\n    test_sql_operators_precedence(client, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 163, in test_sql_operators_precedence\n    raise select_err\n  File \"/mint/run/core/s3select/sql_ops.py\", line 161, in test_sql_operators_precedence\n    test_sql_expressions(client, json_testfile, tests, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 89, in test_sql_expressions\n    input_serialization, output_serialization, tests, log_output)\n  File \"/mint/run/core/s3select/sql_ops.py\", line 74, in test_sql_expressions_custom_input_output\n    client.remove_object(bucket_name, object_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1171, in remove_object\n    object_name=object_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1984, in _url_open\n    region = self._get_bucket_region(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1919, in _get_bucket_region\n    region = self._get_bucket_location(bucket_name)\n  File \"/usr/local/lib/python3.6/dist-packages/minio/api.py\", line 1950, in _get_bucket_location\n    headers=headers)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/poolmanager.py\", line 336, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 828, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 828, in urlopen\n    **response_kw\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 828, in urlopen\n    **response_kw\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/connectionpool.py\", line 805, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.6/dist-packages/urllib3/util/retry.py\", line 439, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='minio-dev7.minio.io', port=31418): Max retries exceeded with url: /s3select-test-4f49adf8-e091-46ba-bbf2-8f07be81b716?location= (Caused by ResponseError('too many 503 error responses',))\n",
  "status": "FAIL"
}
(3/15) Running security tests ... done in 0 seconds

Executed 3 out of 15 tests successfully.
```
<a name="20dd3ac0-4de8-4613-a4a6-8abfd10765f6"></a>
### 9981-bc0a58e/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:31123
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e99eda90cd1f:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 8 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 2 minutes and 57 seconds
(4/15) Running aws-sdk-ruby tests ... done in 16 seconds
(5/15) Running awscli tests ... done in 2 minutes and 48 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 2 minutes and 58 seconds
(8/15) Running minio-dotnet tests ... done in 1 minutes and 34 seconds
(9/15) Running minio-go tests ... done in 4 minutes and 51 seconds
(10/15) Running minio-java tests ... FAILED in 5 minutes and 19 seconds
{
  "name": "minio-java",
  "function": "removeObjects(RemoveObjectsArgs args)",
  "duration": 1256,
  "status": "FAIL",
  "error": "org.simpleframework.xml.core.ValueRequiredException: Unable to satisfy @org.simpleframework.xml.Element(name=\"Key\", type=void.class, data=false, required=true) on field 'name' private java.lang.String io.minio.messages.DeletedObject.name for class io.minio.messages.DeletedObject at line 2 >>> [io.minio.Xml.unmarshal(Xml.java:68), io.minio.MinioClient.deleteObjects(MinioClient.java:7354), io.minio.MinioClient$1$1.populate(MinioClient.java:2945), io.minio.MinioClient$1$1.hasNext(MinioClient.java:2977), FunctionalTest.removeObjects(FunctionalTest.java:1395), FunctionalTest.removeObjects_test1(FunctionalTest.java:1534), FunctionalTest.runTests(FunctionalTest.java:3812), FunctionalTest.main(FunctionalTest.java:4056)]"
}
(10/15) Running minio-js tests ... done in 2 minutes and 37 seconds
(11/15) Running minio-py tests ... done in 9 minutes and 54 seconds
(12/15) Running s3cmd tests ... done in 1 minutes and 24 seconds
(13/15) Running s3select tests ... done in 1 minutes and 4 seconds
(14/15) Running security tests ... done in 1 seconds

Executed 14 out of 15 tests successfully.
```
<a name="8bd08ca2-8960-4b39-b12d-c943fe782907"></a>
### 9981-bc0a58e/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev3.minio.io:31800
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 7a0ca4e7d701:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 12 seconds
(2/15) Running aws-sdk-java tests ... done in 3 seconds
(3/15) Running aws-sdk-php tests ... done in 48 seconds
(4/15) Running aws-sdk-ruby tests ... done in 6 seconds
(5/15) Running awscli tests ... done in 2 minutes and 16 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 1 minutes and 27 seconds
(8/15) Running minio-dotnet tests ... done in 2 minutes and 47 seconds
(9/15) Running minio-go tests ... FAILED in 2 minutes and 31 seconds
{
  "args": {
    "bucketName": "minio-go-test-vudxlmdwtbd5rfis",
    "objectName": "4153pu9uyl65royv5ditq3yad009bd",
    "objectPrefix": "",
    "recursive": "true"
  },
  "duration": 4666,
  "error": "The bucket you tried to delete is not empty. You must delete all versions in the bucket.",
  "function": "RemoveObjectsWithOptions(bucketName, objectsCh, opts)",
  "message": "Cleanup failed",
  "name": "minio-go: testRemoveObjectsWithOptions",
  "status": "FAIL"
}
(9/15) Running minio-java tests ... done in 5 minutes and 50 seconds
(10/15) Running minio-js tests ... done in 1 minutes and 7 seconds
(11/15) Running minio-py tests ... done in 3 minutes and 40 seconds
(12/15) Running s3cmd tests ... done in 23 seconds
(13/15) Running s3select tests ... done in 7 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/api-datatypes.go
+++ b/cmd/api-datatypes.go
@@ -22,7 +22,7 @@ import (
 
 // DeletedObject objects deleted
 type DeletedObject struct {
-	DeleteMarker          bool   `xml:"DeleteMarker"`
+	DeleteMarker          bool   `xml:"DeleteMarker,omitempty"`
 	DeleteMarkerVersionID string `xml:"DeleteMarkerVersionId,omitempty"`
 	ObjectName            string `xml:"Key,omitempty"`
 	VersionID             string `xml:"VersionId,omitempty"`

--- a/cmd/bucket-handlers.go
+++ b/cmd/bucket-handlers.go
@@ -376,6 +376,13 @@ func (api objectAPIHandlers) DeleteMultipleObjectsHandler(w http.ResponseWriter,
 		return
 	}
 
+	// Before proceeding validate if bucket exists.
+	_, err := objectAPI.GetBucketInfo(ctx, bucket)
+	if err != nil {
+		writeErrorResponse(ctx, w, toAPIError(ctx, err), r.URL, guessIsBrowserReq(r))
+		return
+	}
+
 	deleteObjectsFn := objectAPI.DeleteObjects
 	if api.CacheAPI() != nil {
 		deleteObjectsFn = api.CacheAPI().DeleteObjects

--- a/cmd/gateway/azure/gateway-azure.go
+++ b/cmd/gateway/azure/gateway-azure.go
@@ -462,6 +462,8 @@ func azureToObjectError(err error, params ...string) error {
 
 func azureCodesToObjectError(err error, serviceCode string, statusCode int, bucket string, object string) error {
 	switch serviceCode {
+	case "ContainerNotFound":
+		err = minio.BucketNotFound{Bucket: bucket}
 	case "ContainerAlreadyExists":
 		err = minio.BucketExists{Bucket: bucket}
 	case "InvalidResourceName":
@@ -970,7 +972,10 @@ func (a *azureObjects) DeleteObject(ctx context.Context, bucket, object string, 
 	blob := a.client.NewContainerURL(bucket).NewBlobURL(object)
 	_, err := blob.Delete(ctx, azblob.DeleteSnapshotsOptionNone, azblob.BlobAccessConditions{})
 	if err != nil {
-		return minio.ObjectInfo{}, azureToObjectError(err, bucket, object)
+		err = azureToObjectError(err, bucket, object)
+		if !errors.Is(err, minio.ObjectNotFound{Bucket: bucket, Object: object}) {
+			return minio.ObjectInfo{}, err
+		}
 	}
 	return minio.ObjectInfo{
 		Bucket: bucket,
@@ -983,10 +988,8 @@ func (a *azureObjects) DeleteObjects(ctx context.Context, bucket string, objects
 	dobjects := make([]minio.DeletedObject, len(objects))
 	for idx, object := range objects {
 		_, errs[idx] = a.DeleteObject(ctx, bucket, object.ObjectName, opts)
-		if errs[idx] == nil {
-			dobjects[idx] = minio.DeletedObject{
-				ObjectName: object.ObjectName,
-			}
+		dobjects[idx] = minio.DeletedObject{
+			ObjectName: object.ObjectName,
 		}
 	}
 	return dobjects, errs

